src/sampling/fibonacci.jl

```diff
@@ -0,0 +1,50 @@
+"""
+	FibonacciSampling(n)
+
+Generate samples on a sphere using the Fibonacci lattice method.
+The parameter `n` specifies the number of points to generate.
+In the regular Fibonacci lattice method, the number ϕ is the golden ratio ((1 + √5)/2),
+but different numbers can be used, preferably irrational.
+
+## Example
+
+Sample a sphere with 1000 points:
+
+```julia
+sample(Sphere((0,0,0), 1), FibonacciSampling(100))
+
+# sample using π instead of the golden ratio
+sample(Box((0,0),(1,1)), FibonacciSampling(100,π))
+"""
+struct FibonacciSampling{T<:Real} <: ContinuousSamplingMethod
+  n::Int
+  ϕ::T
+
+  function FibonacciSampling(n::Int, ϕ::T) where {T<:Real}
+    if n ≤ 0
+      throw(ArgumentError("Size must be positive"))
+    end
+    new{T}(n, ϕ)
```

Comment:
```suggestion
    new(n, ϕ)
```

Reply:
This change of `new{T}(n, ϕ)` to `new(n, ϕ)` is making tests fail.
```
Failed to precompile Meshes [eacbb407-ea5a-433e-ab97-5258b1ca43fa] to "/Users/davibarreira/.julia/compiled/v1.10/Meshes/jl_7xh45D".
ERROR: LoadError: syntax: too few type parameters specified in "new{...}" around /Users/davibarreira/Documents/GitHub/Forks/Meshes.jl/src/sampling/fibonacci.jl:1
```

Reply:
Interesting I thought that this T parameter wouldn't change anything. We usually call `new` without type parameters in other inner constructors.

Reply:
@eliascarv do you know why T is important in this context?

Reply:
> @eliascarv do you know why T is important in this context?

@juliohm, this is not based on any documented statement, but it seems all other uses of _unparametrized_ `new` within the Meshes codebase are inside inner constructor definitions that themselves are fully parametrized. I believe `new` can only automatically extract the type params not from the arguments it is passed to but from the function signature it is called from:
![image](https://github.com/user-attachments/assets/af61c0d4-f0fb-431a-a648-d072e84143ec)

Reply:
Thank you @disberd ! That explains it. @davibarreira can you please add the type parameter to the function definition like we do in the rest of the codebase and remove it from the new call?

Reply:
No major reason. Just trying to be consistent with most of the other definitions. Let's leave it as is, no problem.